docs/_posts/2024/2024-03-31-scala-io-nantes.md

```diff
@@ -0,0 +1,110 @@
+
+# Focus on Johanna's and Mehdi's talk
+
+Starting with a few dance steps, Johanna and Mehdi presented [Kapoeira](https://github.com/lectra-tech/kapoeira/). 
+Based on cucmber-scala, Kapoeira is an open source library which implements acceptance tests on kafka streams. 
+It provides a simple language based on Gherkin standards (Given - When - Then) so an automated test can be written by a QA, a PO, or a developer.
+If you're looking for high level automated test on kafka stream, you should definitely watch this video.   
+Plus you will love many references about La Cité de la Peur, a classic french comedy.
+
+🇬🇧🌴  ***Youpi dansons la Kapoeira en testant nos Kafka streams*** 🕺 💃<br/>
+[![🌴 Youpi dansons la Kapoeira en testant nos Kafka streams 🕺 💃](https://img.youtube.com/vi/BUQFj2jrGj8/0.jpg){: width="600"}](https://www.youtube.com/watch?v=BUQFj2jrGj8&list=PLjkHSzY9VuL8we5GaQgqWrZSwiKtLM7Cb&index=5)
+
+# Focus on Sophie's talk
+
+If you're not completely comfortable at explaining the concept of contravariance, you should checkout Sophie's talk. She explains this concepts in a very clear way with simple examples.
+She gives practical use cases, using an `Animal` type with concrete sub-types.
+As a prerequiste you should be comfortable with **substitution**, when you have an `Animal` you can pass an instance of `Dog`,
+and with **type classes**, a trait that takes one or more parameters.
+
+#### Invariance
+
+```
+trait Rescue[A]:
+  def adopt(name: String): A
+```
+There is no +, `Rescue` is invariant in `A`, even though `Dog` is a subtype of `Animal` there is no relation between `Rescue[Dog]` and `Rescue[Animal]`
+
+#### Covariance
+
+```
+trait Rescue[+A]:
+  def adopt(name: String): A
+```
+With that little + `Rescue` is **covariant** in `A` so if `Dog` is subtype of `Animal`, then `Rescue[Dog]` is a subtype of `Rescue[Animal]`
+
+Then you can write
+```
+def adopt(name:String)(using rescue: Rescue[Animal]): Animal = rescue.adpot(name)
+
+var poppy = adopt(name = "Poppy")
+```
+#### Contravariance
+```
+trait Clinic[-A]:
+  def examine(partient: A): String
+```
+With that little - `Clinic` is **contravariant** in `A` so if `Dog` is a subtype of `Animal` then `Clinic[Animal]` is subtype of `Clinic[Dog]`, here the relationship is inverted 
+Then you can write
+```
+def examine(dog:Dog)(using clinic:Clinic[Dog]): String = clinic.eximaine(dog)
+
+val medor = Dog("Médor", bred = DogBred.Labrador)
+
+val examintionReport = examine(medor)(using sumon[Clinic[Animal]])
+```
+
+🇬🇧 ***Contravariance: intuition building from first principles*** <br/>
+[![Contravariance: intuition building from first principles](https://img.youtube.com/vi/A7t3b0kymFM/0.jpg){: width="600"}](https://www.youtube.com/watch?v=A7t3b0kymFM)
+
+# Focus on Olya's talk
+
+Olya shared her experience on Ukraine's Scala community building and about the book club organized on the basis of Intelias, where I currently work. 
```

Comment:
Proposition de formulation : Olya shares her exploration of the Ukrainian Scala community and reflects on the lessons learned in the process of community building while setting up a book club within her company Intellias.